### PR TITLE
[CodeCompletion][Sema] Don't filter out any viable solutions when solving for code completion

### PR DESCRIPTION
**Pull request by nathawes**
Resolves rdar://problem/73282163

### CODE REVIEW

**Comment by nathawes** [comment]
@swift-ci please test

File: include/swift/Sema/ConstraintSystem.h [hunk 1]
```diff
@@ -2766,7 +2766,8 @@ class ConstraintSystem {
   void
   filterSolutions(SmallVectorImpl<Solution> &solutions,
                   bool minimize = false) {
-    if (solutions.size() < 2)
+    if (solutions.size() < 2 ||
+        solutions[0].getConstraintSystem().isForCodeCompletion())
```
Comment (xedin) [inline include/swift/Sema/ConstraintSystem.h:2770]: Nit: This is method on constraint system itself so you could simply use `isForCodeCompletion()`.

**Review by xedin — COMMENTED** [review]
I understand the intention here but I'm worried that this might cause "too complex" issues within larger expressions because we'd get all of the partial solutions that have to be merged together so there is going to be a significant memory overhead... What would make me feel better if you could add a couple of examples of expressions with operators + code completion.

**Comment by nathawes** [comment]
@swift-ci please test

**Review by xedin — APPROVED** [review]

**Comment by nathawes** [comment]
@swift-ci please test

**Comment by nathawes** [comment]
@swift-ci please smoke test Linux

**Comment by nathawes** [comment]
@swift-ci smoke test Linux

**Comment by nathawes** [comment]
@swift-ci please test and merge

**Comment by nathawes** [comment]
@swift-ci please test and merge

**Comment by nathawes** [comment]
@swift-ci please test and merge

**Comment by nathawes** [comment]
@swift-ci please test and merge

**Comment by nathawes** [comment]
@swift-ci please smoke test Linux

**Pull request merged**